// @aws-amplify/core/lib/Platform/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAmplifyUserAgent = exports.Platform = void 0;
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
var version_1 = require("./version");
var BASE_USER_AGENT = "aws-amplify/".concat(version_1.version);
exports.Platform = {
    userAgent: "".concat(BASE_USER_AGENT, " js"),
    product: '',
    navigator: null,
    isReactNative: false,
};
if (typeof navigator !== 'undefined' && navigator.product) {
    exports.Platform.product = navigator.product || '';
    exports.Platform.navigator = navigator || null;
    switch (navigator.product) {
        case 'ReactNative':
            exports.Platform.userAgent = "".concat(BASE_USER_AGENT, " react-native");
            exports.Platform.isReactNative = true;
            break;
        default:
            exports.Platform.userAgent = "".concat(BASE_USER_AGENT, " js");
            exports.Platform.isReactNative = false;
            break;
    }
}
var getAmplifyUserAgent = function (content) {
    return "".concat(exports.Platform.userAgent).concat(content ? content : '');
};
exports.getAmplifyUserAgent = getAmplifyUserAgent;
